### ISSUE TITLE
Updating an attachment via the webservice does not delete the old file

### ISSUE THREAD

**Issue by rsoulard-prolaser** [PrestaShop/PrestaShop#27690]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Updating an attachment via the webservice does not delete the old file.

In my environment (Apache/2.4.52, php 7.4.25), it causes a warning type error:
`PHP message: [PHP Warning #2] Use of undefined constant PS_DOWNLOAD_DIR - assumed 'PS_DOWNLOAD_DIR' (this will throw an Error in a future version of PHP) (/.../classes/webservice/WebserviceSpecificManagementAttachments.php, line 330)PHP message: [PHP Warning #2] unlink(PS_DOWNLOAD_DIR04fdcc56393f60923607464740142d1a577931c9): No such file or directory (/.../classes/webservice/WebserviceSpecificManagementAttachments.php, line 330)`

It is fixed on the `develop` branch but it must be fixed also on versions `1.7.8.x`

### Expected behavior

The old file should be deleted

### Steps to reproduce

1. Go to BO
2. Add an attachment
3. The file appears in the `/download` directory
4. Update this attachment via webservice
5. The file is still in the `/download` directory

### PrestaShop version(s) where the bug happened

1.7.8.0

**Comment by hibatallahAouadni** [comment]
Hello @rsoulard-prolaser 

In my case, the update of file from WS didn't work and it always return an empty response, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/154483666-945817f3-ff8b-4d74-a558-22fd8d7aa0d0.png)

And when I check the `my_shop/download` I can't see the new updated file.
Ping @PrestaShop/qa-team can someone please try to reproduce the issue :pray: 

Thanks!

**Comment by rsoulard-prolaser** [comment]
Hello @hibatallahAouadni
The url is not correct, for update an attachment, you must use this url:
`http://prestashop/prestashop_1.7.8.3/api/attachments/file/1?ps_method=PUT`
Thanks

**Comment by rsoulard-prolaser** [comment]
Hello @hibatallahAouadni,
Were you able to test again?
My PR corrects a typo and it would be nice if it was merged for version 1.7.8.6
Thanks

**Comment by khouloudbelguith** [comment]
Hi @rsoulard-prolaser,

Sorry for the late response.
I manage to reproduce the issue with PS1785
I attached a screen record to show the difference between updating a file via API or via BO
Update a file via API => the old file is not deleted and the new file is well added
Update a file from BO => the old file is replaced by the new file

I attached a screen record

https://user-images.githubusercontent.com/16067358/160885177-37745a56-0860-4dd8-bd97-cedfb4133791.mp4

I tried with PS1778, I couldn't find the attachment
![image](https://user-images.githubusercontent.com/16067358/160886034-7fd0e0ae-90c7-404e-82b0-d559547f9ea2.png)
In the PS1785
![image](https://user-images.githubusercontent.com/16067358/160886193-2979bc65-e2b8-46b2-b06e-bdb3abd9bc34.png)

This is a new feature in PS1780
https://github.com/PrestaShop/PrestaShop/pull/1272

so I can confirm that this behavior is indeed a bug.
Thanks for the PR!
